sdk/storage/azblob/hashing/hashing_reader.go

```diff
@@ -0,0 +1,46 @@
+package hashing
```

Comment:
Do we think there's enough content here (or future content) to warrant placing this in its own package?  Maybe it can just live in the `blob` package.  Not advocating for either at present but want to start the discussion.

Reply:
I think it may well have room in the `blob` package. I can't see any particularly good reason why it would be bad to put it there, outside of a dependency loop, but all of the specialized blob packages already reference the core `blob` package.

Reply:
As for future substance, it is likely we'll utilize the ability to concatenate hashes together for use in the `chunkwriting.go` file in the `blockblob` package, both for upload and download.

Reply:
I think it would be best to move this to the `blob` package. That way, we don't have to add more public surface area (e.g. `ReadWrapper`) that's not really required to be public.

Reply:
> I think it would be best to move this to the `blob` package. That way, we don't have to add more public surface area (e.g. `ReadWrapper`) that's not really required to be public.

ReadWrapper would have to be an exported type anyway as it's used by all 3 specialized blob types, which are separate packages; unless we wanted to incur code dupe.

Reply:
As per offline discussion, hashing type option has been moved to `internal\exported` and the rest to `internal\shared`.